planner/core/logical_plan_builder.go

```diff
@@ -2253,6 +2253,10 @@ func (r *correlatedAggregateResolver) resolveSelect(sel *ast.SelectStmt) (err er
 	if err != nil {
 		return err
 	}
+	// do not use cache when for update read
+	if isForUpdateReadSelectLock(sel.LockInfo) {
+		useCache = false
```

Comment:
after debug, it seems `useCache = false` only not-use-cache, but it still refill cache..

so in Access Path Selection will happen in `correlatedAggregateResolver` and directly use cache in later buildProjection

Reply:
Seems `useCache = false` only cache sub-sub-query when there is a select stmt inside from clause.

Reply:
I added some cases for sub queries, the cache problem is also related with rewrite, it may be solved together with it.